pythnet-price-feeds/cosmwasm.md

```diff
@@ -6,9 +6,30 @@ description: Consume Pyth Network prices in applications on Cosmwasm
 
-## Updating Price Feeds
+## Price Feeds
+Pythnet price feeds uses an on-demand price update model, where users are responsible for posting price updates on-chain when needed. The latest price can be fetched from a price service, which is a permissionless centralised service. We recommend protocols to run their own price-service. 
+Pythnet publishes prices for two kinds of feeds: 
+1. Stable price feeds are the most reliable price feeds.
+2. Edge price feeds comes with the latest and greatest features of Pyth price feeds as soon as they are available. 
 
-The mechanism by which price feeds are updated on Cosmwasm is explained [here](./pythnet-price-feeds.md). The [pyth-common-js](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/cosmwasm/sdk/js) can be used to fetch the latest price feed data which then can be passed to the `ExecuteMsg` on-chain function.
+Mainnet only supports the stable version of price feeds whereas testnet supports both. 
+
+To learn more about the mechanism by which price feeds are updated on Cosmwasm you can follow this [link](./pythnet-price-feeds.md). The [pyth-common-js](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/cosmwasm/sdk/js) can be used to fetch the latest price feed data which then can be passed to the `ExecuteMsg` on-chain function.
+
+### Price Service Public Endpoints
+| Price feeds | url                             |
+| ------- | ------------------------------- |
+| stable | https://xc-mainnet.pyth.network |
+| edge | https://xc-testnet.pyth.network |
+
+```
+# Example API call
+$ curl https://xc-testnet.pyth.network/api/latest_vaas?ids[]=0xf9c0172ba10dfa4d19088d94f5bf61d3b54d5bd7483a322a982e1373ee8ea31b
+
+# Example Response
+["AQAAAAABALNY33VN+Pxh+vh02T9mAjCkNlug8K7qYTEPa+vU/hcSP+IGYDm75QCZ2oVdYdV45tXmsnAqnQwnybTt5QqOX8wBZG4AGAAAAAAAGqJ4OdZBsHdDwMtfaMUfjNMdLAdivsANxvzSVDPvGrW2AAAAAAMOaTIBUDJXSAADAAEAAQIABQCdHNsaXh40VtKXfuDT1wdlI58IpChVuVCP1HnhXG3E0f7s9VN3DZsQll+Ptkdx6T9WkKGC7cMr5KMjbgyqpuBYGgAAAAcxcTjgAAAAAAG6Nuf////4AAAABy+1l2gAAAAAAS3wagEAAAABAAAAAgAAAABkbgAYAAAAAGRuABQAAAAAZG4AEgAAAAcxWUAQAAAAAAEH63AAAAAAZG4AFGogZxwOP4yyGc4/RuWuCWpPL9+TbSvU2okl9wCH1R3YMAKUeVmHlykONjihcSwpveI2fQ7KeU93iyW1pHLxkt4AAAACoO+6gAAAAAAAmJaA////+AAAAAKgDbl4AAAAAACW+rcBAAAAAQAAAAIAAAAAZG4AGAAAAABkbgAXAAAAAGRuABQAAAACoOKsoQAAAAAAw+FuAAAAAGRuABco/gXScIxlcRgqfJ0f9FeiIbRl7fXqmvE3P5Vi0WuNFfnAFyuhDfpNGQiNlPW/YdO1TVvXSDoyKpguE3PujqMbAAACbg2tzH8AAAAAHz3sv/////gAAAJuPkE5QAAAAAAkkZ1wAQAAAAEAAAACAAAAAGRuABgAAAAAZG4AFwAAAABkbgAVAAACbg13O78AAAAAJKtrvwAAAABkbgAXizjbcA6LNGQOaB7Jpz6JYIvaKUFVR6Ik+WWFGStLnceUvOSu6I/fpbWNgQkL1rN4Rxf6bfhUGdnwRDO7PWFdXAAAAAAFAlNUAAAAAAAA9ED////4AAAAAAUHGlgAAAAAAAEesgEAAAABAAAAAgAAAABkbgAYAAAAAGRuABcAAAAAZG4AFAAAAAAFAlNUAAAAAAAA9EAAAAAAZG4AFztpo88HVkbF/YFItwW4EH5hoaJT1dioQ1Xctiiz8dEgMXdeHWiXEp6KhO66l1d4+1ABW4gDnpvBQLvYOWlKwK4AAAAAAGyBWAAAAAAAABOK////+AAAAAAAbHdcAAAAAAAAD4ABAAAAAQAAAAIAAAAAZG4AGAAAAABkbgAYAAAAAGRuABUAAAAAAGyBWAAAAAAAABOKAAAAAGRuABU="]
+
+```
```

Comment:
kind of seems like this example query should go in the Price Service section of the docs?